fmt/container.h

```diff
@@ -0,0 +1,82 @@
+/*
+ Formatting library for C++ - standard container utilities
+
+ Copyright (c) 2012 - 2016, Victor Zverovich
+ All rights reserved.
+
+ For the license information refer to format.h.
+ */
+
+#ifndef FMT_CONTAINER_H_
+#define FMT_CONTAINER_H_
+
+#include "format.h"
+
+namespace fmt {
+
+namespace internal {
+
+/**
+  \rst
+  A "buffer" that appends data to a standard container (e.g. typically a
+  ``std::vector`` or ``std::basic_string``).
+  \endrst
+ */
+template <typename Container>
+class ContainerBuffer : public Buffer<typename Container::value_type> {
+ private:
+  Container& container_;
+
+ protected:
+  virtual void grow(std::size_t size) FMT_OVERRIDE {
+    container_.resize(size);
+    this->ptr_ = &container_[0];
+    this->capacity_ = size;
+  }
+
+ public:
+  explicit ContainerBuffer(Container& container) : container_(container) {
+    this->size_ = container_.size();
+    if (this->size_ > 0) {
+      this->ptr_ = &container_[0];
+      this->capacity_ = this->size_;
```

Comment:
> ... I'd omit the zero check altogether.

The point of the test is to avoid assuming that it is safe to initialize `ptr_` (thus assuming `&container_[0]` is always safe) before being sure it is (i.e. after the `container_.resize()` call). Although admittedly it should be fine with standard containers with most compilers, but we cannot assume `Container` type is standard here, or derived from a standard.

> Looks like capacity will be uninitialized if size is zero.

Doesn't `fmt::Buffer`'s default constructor set `capacity_` to zero?

> Could you add a test case that catches this?

You mean like `TEST(BasicContainerWriterTest, String)`, which already tests that case unless I misunderstood you, but explicitly for `ContainerBuffer`?

> Does it compile?

The corrected example compiles on MSVC2015 SP3. Tested before committing of course :)
I cannot confirm it compiles just fine on other configs though, or that it is even C++98 compliant, like it's supposed to be for fmt. Do note however I followed the same example pattern found in the documentation of the `FMT_VARIADIC` macro, which has a `void` return type as well.

Also, please note MSVC2015 SP3 is the only compiler I have access to ATM since I have quite limited access and bandwidth to Internet.